src/wavm/mpi.cpp

```diff
@@ -1355,7 +1355,8 @@ WAVM_DEFINE_INTRINSIC_FUNCTION(env,
     int* coordsArray =
       Runtime::memoryArrayPtr<int>(ctx.memory, (Uptr)coords, (Uptr)maxdims);
 
-    ctx.world.getCartesianRank(ctx.rank, dimsArray, periodsArray, coordsArray);
+    ctx.world.getCartesianRank(
+      ctx.rank, maxdims, dimsArray, periodsArray, coordsArray);
```

Comment:
Changes ported to allow compiling against latest Faabric. Should be able to revert once Faabric PR merged in.

Reply:
Sure! I am working on it now.